pkg/linear-pools/contracts/tetu-linear-pool/TetuShareValueHelper.sol

```diff
@@ -26,21 +25,22 @@ import "./interfaces/ITetuStrategy.sol";
 
 contract TetuShareValueHelper {
     using SafeERC20 for IERC20;
-    using Math for uint256;
+    using FixedPoint for uint256;
 
-    function _getTokenRate(address wrappedTokenAddress) internal view returns (uint256) {
-        // Since there's fixed point divisions and multiplications with rounding involved, this value might
-        // be off by one. We add one to ensure the returned value will always be enough to get `wrappedAmount`
-        // when unwrapping. This might result in some dust being left in the Rebalancer.
+    uint256 private immutable _defaultRate = FixedPoint.ONE;
 
+    function _getTokenRate(address wrappedTokenAddress) internal view returns (uint256) {
         uint256 wrappedTokenTotalSupply = _getWrappedTokenTotalSupply(wrappedTokenAddress);
         if (wrappedTokenTotalSupply == 0) {
-            return 0;
+            // Since Tetu TokenRate is basically 1 + yield, 1 is a good default value when there's no supply (yield = 0)
+            return _defaultRate;
         } else {
             uint256 underlyingBalanceInVault = _getUnderlyingBalanceInVault(wrappedTokenAddress);
             uint256 strategyInvestedUnderlyingBalance = _getStrategyInvestedUnderlyingBalance(wrappedTokenAddress);
-            uint256 balance = Math.add(underlyingBalanceInVault, strategyInvestedUnderlyingBalance);
-            return FixedPoint.divDown(balance, wrappedTokenTotalSupply);
+            uint256 balance = underlyingBalanceInVault.add(strategyInvestedUnderlyingBalance);
+            // Since there's fixed point divisions and multiplications with rounding involved, this value might
+            // be off by one, so we add 1 to the rate to make sure the conversion will result in enough main tokens
+            return balance.divDown(wrappedTokenTotalSupply).add(1);
```

Comment:
Added 1 to the _getTokenRate calculation because _getTokenRate has a lot of divisions and it's off by 1 in the unit tests.

Reply:
This imprecision seems to stem from the way the rate is set in the `MockTetuSmartVault`. In short, it does a calculation like this:

```
underlying = total / 3
invested = 2 * total / 3
```

The problem is, `underlying` and `invested` don't actually add up to `total`, because of rounding errors. So when the rate is calculated later, it is based on an incorrect `total`. A better approach might be one like this:

```
underlying = total / 3
invested = total - underlying
```

Reply:
This imprecision seems to stem from the way the rate is set in the `MockTetuSmartVault`. In short, it does a calculation like this:

```
underlying = total / 3
invested = 2 * total / 3
```

The problem is, `underlying` and `invested` don't actually add up to `total`, because of rounding errors. So when the rate is calculated later, it is based on an incorrect total sum. A better approach might be one like this:

```
underlying = total / 3
invested = total - underlying
```

Reply:
Man, I haven't seen that! You're right!